geth/api/backend.go

```diff
@@ -77,120 +80,145 @@ func (m *StatusBackend) IsNodeRunning() bool {
 }
 
 // StartNode start Status node, fails if node is already started
-func (m *StatusBackend) StartNode(config *params.NodeConfig) (<-chan struct{}, error) {
+func (m *StatusBackend) StartNode(config *params.NodeConfig) error {
 	m.Lock()
 	defer m.Unlock()
+	return m.startNode(config)
+}
 
-	if m.nodeReady != nil {
-		return nil, node.ErrNodeExists
-	}
-
-	nodeStarted, err := m.nodeManager.StartNode(config)
-	if err != nil {
-		return nil, err
+func (m *StatusBackend) startNode(config *params.NodeConfig) error {
+	if err := m.nodeManager.StartNode(config); err != nil {
+		switch err.(type) {
+		case node.RPCClientError:
+			signal.Send(signal.Envelope{
+				Type: signal.EventNodeCrashed,
+				Event: signal.NodeCrashEvent{
+					Error: node.ErrRPCClient.Error(),
```

Comment:
One more generic question: why is `Error` field in the signal a `string` type and not an `error` interface?
Is there any background story for that?

Reply:
maybe because there is no automatic marshaling of errors

Reply:
Hmm, maybe we should make a custom JSON marshaller for `NodeCrashEvent` and enforce types by making `Error` field of an `error` type: https://play.golang.org/p/7ZpQWTOv4kG
We can make a bounty out of it.